core/src/main/java/org/opensearch/sql/data/model/ExprCollectionValue.java

```diff
@@ -73,7 +73,7 @@ public boolean equal(ExprValue o) {
    */
   @Override
   public int compare(ExprValue other) {
-    return Integer.compare(valueList.size(), other.collectionValue().size());
+    return equal(other) ? 0 : 1;
```

Comment:
Is this identical implementation in `ExprTupleValue` needed twice?

Reply:
I reverted this in d6f68b97.